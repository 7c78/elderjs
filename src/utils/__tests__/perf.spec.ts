import perf from '../perf';

jest.mock('perf_hooks', () => {
  return {
    PerformanceObserver: function PerformanceObserver() {
      return {
        observe: jest.fn(),
        disconnect: jest.fn(),
      };
    },
    performance: {
      mark: jest.fn(),
      measure: jest.fn(),
      clearMarks: jest.fn(),
    },
  };
});

function MockPage() {
  this.uid = 'xxxxxxxx';
  this.htmlString = '';
}

describe('#perf', () => {
  it('works', () => {
    const mockPage = new MockPage();
    // mutate
    perf(mockPage);

    mockPage.perf.start('test');
    mockPage.perf.end('test');
    mockPage.perf.stop();

    expect(mockPage).toMatchSnapshot();
  });
});
