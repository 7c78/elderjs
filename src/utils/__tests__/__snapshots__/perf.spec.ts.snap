// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#perf works 1`] = `
MockPage {
  "htmlString": "",
  "perf": Object {
    "end": [Function],
    "start": [Function],
    "stop": [Function],
    "timings": Array [],
  },
  "uid": "xxxxxxxx",
}
`;
